src/LayerEditor/ui/tools/undo.py

```diff
@@ -52,4 +76,30 @@ def clear():
     """Clear extra info in this extension"""
     global __savepoint
     stack().clear()
-    __savepoint = None
\ No newline at end of file
+    __savepoint = None
+
+
+def undoable_prepend(generator):
+    ''' Decorator which creates a new undoable action type.
```

Comment:
`@undo.undoable` appends action, this prepends the action to the stack. It was needed for the callbacks which were removed. Now I removed this method as well since its not used.